Assets/Scripts/Augment/AugmentImplementations/Hat/HatBarrelModel.cs

```diff
@@ -19,42 +19,36 @@ public class HatBarrelModel : MonoBehaviour
     private List<GameObject> ammunition = new List<GameObject>();
 
     private int magazineSize = 5;
-    private int remainingBullets;
 
-    void Start()
+    public void OnInitialize(int magazine)
     {
-        remainingBullets = magazineSize;
+        magazineSize = magazine;
         for (int i = 0; i < magazineSize; i++)
         {
             ammunition.Add(Instantiate(hatPrefab, ammunitionHolder.transform));
             ammunition[i].transform.Translate(new Vector3(0f, bulletHeigth * i, 0f));
         }
```

Comment:
This whole class feels like something that could work for most barrels. Perhaps we could refactor it (later)?

Reply:
Yes, i would very much like to so!

I'll do it when i refator attachmentPoints and outputs